rust/src/index/vector/kmeans.rs

```diff
@@ -54,6 +56,11 @@ pub async fn train_kmeans(
     mut rng: impl Rng,
 ) -> Result<Float32Array> {
     let num_rows = array.len() / dimension;
+    if num_rows < k as usize {
+        return Err(crate::Error::Index(format!(
+            "KMeans: can not train {k} centroids with {num_rows} vectors"
```

Comment:
A more helpful message here would give the user some guidance on what `k` should be

Reply:
updated.